framework/src/modules/chain/logic/block.js

```diff
@@ -232,24 +234,27 @@ class Block {
 				delete block[key];
 			}
 		});
-
 		const report = this.scope.schema.validate(block, Block.prototype.schema);
-
 		if (!report) {
 			throw `Failed to validate block schema: ${this.scope.schema
 				.getLastErrors()
 				.map(err => err.message)
 				.join(', ')}`;
 		}
-
 		try {
-			for (let i = 0; i < block.transactions.length; i++) {
-				block.transactions[i] = this.scope.transaction.objectNormalize(
-					block.transactions[i]
-				);
+			const transactionResponses = modules.processTransactions.validateTransactions(
+				block.transactions
+			);
+			const invalidTransactionResponse = transactionResponses.find(
+				transactionResponse =>
+					transactionResponse.status !== TransactionStatus.OK
+			);
+			if (invalidTransactionResponse) {
+				throw invalidTransactionResponse.errors;
 			}
-		} catch (e) {
-			throw e;
+		} catch (errors) {
+			const error = Array.isArray(errors) && errors.length > 0 ? errors[0] : errors;
```

Comment:
Why only first error here if it's an array?

Reply:
I think originally it was only returning one error